tikv-client/src/main/java/com/pingcap/tikv/codec/Codec.java

```diff
@@ -510,10 +511,13 @@ static DateTime fromPackedLong(long packed, DateTimeZone tz) {
       int minute = (hms >> 6) & ((1 << 6) - 1);
       int hour = hms >> 12;
       int microsec = (int) (packed % (1 << 24));
-      return new DateTime(
+
+      LocalDateTime localDateTime = new LocalDateTime(
           year, month, day, hour,
-          minute, second, microsec / 1000,
-          tz);
+          minute, second, microsec / 1000);
```

Comment:
1. Maybe you can put normal code path into a try-catch and only when exception happens then do the heavy path. I guess that might save you some time.
2. Please test the behavior of TiDB itself. For example, Timestamp type involves a TZ conversion from / to UTC. If time doesn't "jump" like in our solution, please write it down in document.

Reply:
Timestamp is always UTC (stored in UTC and decoded as UTC). It should not be affected by this issue in any code path. DateTime is the source of bug since it needs to be decoded as different timezones according to our implementation.